### PR TITLE
Add community collaboration interest notifications.

### PR DESCRIPTION
**Pull request by alstojanovic**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

File: src/services/domain/builders/community-collaboration-interest/community.collaboration.interest.notification.builder.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      emailFrom: 'info@alkem.io',
+      user: {
+        name: user.displayName,
+        email: user.email,
+      },
+      recipient: {
+        firstname: recipient.firstName,
+        email: recipient.email,
+      },
+
```
Comment (hero101) [inline src/services/domain/builders/community-collaboration-interest/community.collaboration.interest.notification.builder.ts:72]: empty line

**Review by hero101 — CHANGES_REQUESTED** [review]
Unable to test due to missing migration for user preference.

File: src/common/dto/community.collaboration.interest.payload.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export type CommunityCollaborationInterestPayload = {
```
[the remark this answers is not in the document]
Comment (alstojanovic) [reply src/common/dto/community.collaboration.interest.payload.ts:1]: The server now matches this.

File: src/common/constants/events.ts [hunk 3]
```diff
@@ -8,3 +8,5 @@ export const COMMUNITY_CONTEXT_REVIEW_SUBMITTED =
   'communityContextReviewSubmitted';
 export const ASPECT_CREATED = 'aspectCreated';
 export const COMMENT_CREATED_ON_ASPECT = 'commentCreatedOnAspect';
+export const COMMUNITY_COLLABORATION_INTEREST =
+  'communityCollaborationInterest';
```
[the remark this answers is not in the document]
Comment (alstojanovic) [reply src/common/constants/events.ts:12]: The server now matches this.

File: src/common/dto/community.collaboration.interest.payload.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (hero101) [reply src/common/dto/community.collaboration.interest.payload.ts:1]: > The server now matches this.
The type of `communityName` does not match the type in the server.

File: notifications.yml [hunk 4]
```diff
@@ -191,6 +191,20 @@ recipients:
         - rule:
             type: USER_SELF_MANAGEMENT
             resource_id: <memberID>
+  community_collaboration_interest:
+    - name: admin
```
[the remark this answers is not in the document]
Comment (hero101) [reply notifications.yml:195]: I still do not believe `GLOBAL_ADMIN_COMMUNITY` to be required here.

**Review by hero101 — CHANGES_REQUESTED** [review]
The build is failing.

File: notifications.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (alstojanovic) [reply notifications.yml:195]: Removed that the notifications admin as well.

**Pull request merged**